File: InsertionSort.java
public class InsertionSort {
	
	public static void main(String[] args){
		int[] myArray = new int[]{10,-3,7,2,5,9,4,6};
		
		System.out.println("Initial array.");
		for(int i = 0; i < myArray.length; i++){
			System.out.println(myArray[i]);
		}
		System.out.println();
		
		for(int firstUnsortedIndex = 1; firstUnsortedIndex < myArray.length; firstUnsortedIndex++){
			int newElement = myArray[firstUnsortedIndex];
			int i;
			
			for(i = firstUnsortedIndex; i > 0 && myArray[i-1] > newElement; i--){
				myArray[i] = myArray[i-1];
			}
			myArray[i] = newElement;
		}
		
		System.out.println("Sorted array.");
		for(int i = 0; i < myArray.length; i++){
			System.out.println(myArray[i]);
		}
		
	}
	
}

File: insertIon_sort.go
package main

import "fmt"

func main(){
	dataInput := []int{4, 1, 5, 8, 9, 3, 2, 6, 7}
	fmt.Printf("Unsorted array: %d\n", dataInput)
	InsertionSort(dataInput)
	fmt.Printf("Sorted array: %d\n", dataInput)
}

func InsertionSort(data []int) {
	for i := 1; i < len(data); i++ {
		if data[i] < data[i-1] {
			j := i - 1
			temp := data[i]
			for j >= 0 && data[j] > temp {
				data[j+1] = data[j]
				j--
			}
			data[j+1] = temp
		}
	}
}
